lib: lte_link_control: Remove deprecated (de)init functions (#16629)
nrfconnect/sdk-nrf: doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst

Modem libraries
---------------

* :ref:`lte_lc_readme` library:

  * Removed:

>     * The deprecated :c:func:`lte_lc_init` function.
>       Use of this function can simply be removed without other actions.
>     * The deprecated :c:func:`lte_lc_deinit` function.
>       Use the :c:func:`lte_lc_power_off` function instead.
>     * The deprecated :c:func:`lte_lc_init_and_connect` function.
>       Use the :c:func:`lte_lc_connect` function instead.
>     * The deprecated :c:func:`lte_lc_init_and_connect_async` function.
>       Use the :c:func:`lte_lc_connect_async` function instead.

Reviewer: ```suggestion
    * The :c:func:`lte_lc_init` function.
      All instances of this function can be removed without any additional actions.
    * The :c:func:`lte_lc_deinit` function.
      Use the :c:func:`lte_lc_power_off` function instead.
    * The :c:func:`lte_lc_init_and_connect` function.
      Use the :c:func:`lte_lc_connect` function instead.
    * The :c:func:`lte_lc_init_and_connect_async` function.
      Use the :c:func:`lte_lc_connect_async` function instead.
```
